onnxruntime/core/optimizer/matmul_integer_to_float.cc

```diff
@@ -64,8 +64,7 @@ Status MatMulIntegerToFloatFusion::ApplyImpl(Graph& graph, bool& modified, int g
 
     ORT_RETURN_IF_ERROR(Recurse(mul_node, modified, graph_level, logger));
 
-    if (!graph_utils::IsSupportedOptypeVersionAndDomain(mul_node, "Mul", {7, 13, 14}) ||
-        !graph_utils::IsSupportedProvider(mul_node, GetCompatibleExecutionProviders())) {
```

Comment:
This blocks graph fusion, `IsSupportedProvider` list for FP16 is empty, investigating why

Reply:
🤔 I'm not familiar with `GetCompatibleExecutionProviders`. I'd need to debug through it to see what's in the list, but I suspect the correct thing is to leave the line here and instead figure out why `GetCompatibleExecutionProviders` is missing DML from the list.

Reply:
The graph performs fusion and at this point most operators have default CPU provider listed with respective registered CPU ORT kernels, as "Mul" for FP16 does not have a regeistered kernel(gets converted to FP32 Mul with cast), the "Mul" node has an empty providers list. 
By taking this check out we are able to fuse the nodes.
Although right way would be to have cpu provider listed for the Mul node even if it does not have a cpu registered kernel, which I guess needs more discussion